### PR TITLE
 Added Gulp Support

### LINKED ISSUE #5: Add a simple gulpfile for scss and js
Just a feature request. How about a simple gulpfile for compiling css and js :)? Would be handy.

Browsersync wouldnt hurt as well.

### PR DESCRIPTION
**Pull request by usamamuneerchaudhary**
# Added Gulp Support
- Added the capability to utilize gulp for styles, scripts & images compression and minification.

# Added NPM option
- Added the option for the user to choose if they want to install node packages or not by passing the flag --npm while creating project.

Also fixes #5

### CODE REVIEW

File: theme/GulpFile.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        .pipe(concat('main.css'))
+        .pipe(rename({suffix : '.min'}))
+        .pipe(gulp.dest('assets/css/'))
+});
+
+/**
+ * Images Task
+ */
+gulp.task('images', function () {
+    return gulp.src(['resources/images/**/*'])
```
Comment (azeemhassni) [inline theme/GulpFile.js:25]: What if there are non-image files ?

File: theme/GulpFile.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        .pipe(imgmin())
+        .pipe(gulp.dest('assets/images/'));
+});
+
+/**
+ * Scripts Task
+ */
+gulp.task('scripts', function(){
+    return gulp.src(['resources/libs/**/*.js'])
```
Comment (azeemhassni) [inline theme/GulpFile.js]: Shouldn't we use `resources/js` instead of `libs` ?

File: theme/GulpFile.js (hunk 1 above)
Comment (usamamuneerchaudhary) [reply theme/GulpFile.js:25]: It will compress everything and move that to assets/images directory

**Review by azeemhassni — APPROVED** [review]

**Comment by covertbert** [comment]
So how do you actually include gulp or NPM? The `--npm` flag does not work.

**Comment by azeemhassni** [comment]
@covertbert 
This PR is included in [2.2](https://github.com/azeemhassni/pine/releases/tag/2.2) so make sure you update pine before using `--npm` flag.

**Comment by covertbert** [comment]
@azeemhassni Nice one, cheers mate!

**Pull request merged**